OpenRA.Mods.AS/Traits/World/TintedCellsLayer.cs

```diff
@@ -46,6 +46,9 @@ public class TintedCellsLayerInfo : TraitInfo
 		[Desc("Z offset of the visualization.")]
 		public readonly int ZOffset = 512;
```

Comment:
We already had a ZOffset up here. How is the two any different other than now allowing X and Y too? Is this value used somewhere else? If not it should be removed.

Reply:
Offset (xyz) is a quite different thing compared to ZOffset(or Zsorting). Offset will actually affect the position of the image, while ZOffset decides put who to the top when many images takes one position.

In the gif I showed above,  tinted cell (considered as an image) has the same position as unit, but its ZOffset set to very low so it make sure unit always on the top of tinted cell.

Set the same value into Z axis of Offset will result in the entire cells move downward.

At last, as long as we initialise the ZOffset in 'TintedCell' class, the upper caller will use it to adjust the image.

Reply:
It's used here
![Screenshot_20241103_141543_org_mozilla_firefox_HomeActivity_edit_10925252420883](https://github.com/user-attachments/assets/88407bf3-d3bf-44bb-bca2-cbff8f8e57c8)

Reply:
and at `WorldRenderer` when it is drawing, the zoffset is called
![3](https://github.com/user-attachments/assets/16b8d75a-9389-46d1-a45a-b3ec1debe881)
 to adjust which image on the top.

Reply:
so we should not touch zoffset

Reply:
Yeah, sorry. I had just woken up. Only remembered about the difference while having breakfast.